src/main/java/jenkins/scm/impl/subversion/SubversionSCMSource.java

```diff
@@ -310,6 +311,8 @@ protected SCMRevision retrieve(String unparsedRevision, TaskListener listener) t
             return new SCMRevisionImpl(new SCMHead(base), revision == -1 ? resolvedRevision : revision);
         } catch (SVNException e) {
             throw new IOException(e);
+        } finally {
+            closeSession(repository);
```

Comment:
I am pretty sure this was the cause of the reported leak on [JENKINS-42186](https://issues.jenkins-ci.org/browse/JENKINS-42186) given that successive calls to `retrieve` seem to be able to reproduce the leak.